engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -141,7 +142,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
@rafaelweingartner 
Well the Idea was to tell that CloudRuntimeExceptions should also be sent to the caller(in case of current Cloudstack code), and then the caller will add more context and throw a single exception, which will be encapsulating all the lower level exceptions. Currently if i make RuntimeException a non Runtime one it will break the code at many places. The idea was to do a cleanup bit by bit and once we are at a position to change the RuntimeException to a non runtime we do it. Well at least this is what i thought.